openmc/deplete/abc.py

```diff
@@ -763,6 +773,14 @@ def _get_start_data(self):
         return (self.operator.prev_res[-1].time[-1],
                 len(self.operator.prev_res) - 1)
 
+    def _get_bos_from_batchwise(self, step_index, bos_conc):
+        """Get BOS from criticality batch-wise control
+        """
+        x = deepcopy(bos_conc)
+        # Get new vector after keff criticality control
+        x, root = self._batchwise.search_for_keff(x, step_index)
```

Comment:
Subclasses of the abstract base class are not required to implement `search_for_keff` since it is not defined on the base class at all, let alone an abstract method. Please add a corresponding abstract `search_for_keff(...)` method.